bokehjs/src/coffee/models/transforms/interpolator.coffee

```diff
@@ -0,0 +1,85 @@
+_ = require "underscore"
+Transform = require "./transform"
+p = require "../../core/properties"
+
+class Interpolator extends Transform.Model
+
+  initialize: (attrs, options) ->
+    super(attrs, options)
+    @_x_sorted = []
+    @_y_sorted = []
+    @_sorted_dirty = true
+
+    @bind 'change', () ->
+      console.log 'Changed the configuration'
+      @_sorted_dirty = true
+
+  props: ->
+    return _.extend {}, super(), {
+      x: [ p.Any, '']
+      y: [ p.Any, '']
+      data: [ p.Any, null]
+    }
+
+  sort: (descending = false) ->
+    # Verify that all necessary objects exist...
+    if typeof(@get('x')) != typeof(@get('y'))
+      # ToDo: Throw a resonable error here
+      return
+
+    if typeof(@get('x')) == 'object'
+      if @get('x').length != @get('y').length
+        # ToDo: Throw a resonable error here
+        return
+    else
+      if @get('x').length == 0
+        return
+
+      if @get('y').length == 0
+        return
+
+      if @get('data') == null
+        # ToDo: Throw a resonable error here.
+        return
+
+    # Stop processing this if the dirty flag is not set
+    if(@_sorted_dirty == false)
+      return
+
+    tsx = []
+    tsy = []
+
+    # Populate the tsx and tsy variables correctly depending on the method by which the user populated the interpolation
+    # data.
+    if typeof(@get('x')) == 'string'
+      data = @get('data')
+
+      tsx = data.get_column(@get('x'))
+      tsy = data.get_column(@get('y'))
+    else
+      tsx = @get('x')
+      tsy = @get('y')
```

Comment:
@bryevdv Corrected, I think.

Concerning the question about allow for a string in one parameter and then an array in the other...  I had thought about that as well but was concerned that it would increase the possibility of a length mismatch between the x and the y.  This could also occur if the user changed the source or the array.  At each change, the dirty flag would be raised and the system would then throw an error because one or the other would not be the same length as the other (possibly).  This is because you can not specify all of these in a single call.

However, if the user specifies two arrays, there is no way to change the length of the arrays without throwing an error (if everything works the way I expected it too).

This might mean that the correct thing to do would be to simply have the interpolators return a NA style value in the event that the defining datasets for the interpolation are in a failed state.  

Thoughts?